ChatSX/components/ParseDeepLink/index.ts

```diff
@@ -0,0 +1,101 @@
+import {myClient} from '../../..';
+import {PATH_REGEX, QUERY_REGEX, VALID_URI_REGEX} from '../../constants/Regex';
+import {initAPI} from '../../store/actions/homefeed';
+
+interface DeepLinkRequest {
+  uri: string;
+  userName: string;
+  uuid: string;
+  isGuest: boolean;
+}
+
+interface DeepLinkResponse {
+  success: boolean;
+  errorMessage?: string;
+}
+
+// this function is to check URI is valid or not
+function isValidURI(uri: string): boolean {
+  const regex = VALID_URI_REGEX;
+  return regex.test(uri);
+}
```

Comment:
why this necessary, as this regex check all types of links like `ftp` etc, if we want to add this check, then add check for `https://` links